### DISCUSSION TITLE
Pyright can’t find a submodule when I import the root package

### DISCUSSION THREAD

**Discussion by Pyprohly** [microsoft/pyright#3663] [Q&A] [answered]
```python
# import pkg
import pkg.util.foo
pkg.util.foo.Foo()
```

If I uncomment the first import in the preceding code, Pyright reports that the module `foo` does not exist. Should this happen?

Note I have an explicit import for module `util` in `pkg/__init__.py` which seems to be the cause of the issue.

```text
pkg
├── __init__.py
│   from . import util
│
└── util
    ├── __init__.py
    └── foo.py
        class Foo:
            pass
```

**Comment by erictraut** [comment, marked as answer]
I think this should work without error. Converting to a bug report.
